generated/nifake/nifake_service.cpp

```diff
@@ -644,7 +645,46 @@ namespace nifake_grpc {
         response->set_status(status);
         if (status == 0) {
           response->set_array_out(array_out);
-          response->mutable_array_out()->resize(actual_size-1);
+          nidevice_grpc::trim_trailing_nulls(*(response->mutable_array_out()));
+          response->set_actual_size(actual_size);
+        }
+        return ::grpc::Status::OK;
+      }
+    }
+    catch (nidevice_grpc::LibraryLoadException& ex) {
+      return ::grpc::Status(::grpc::NOT_FOUND, ex.what());
+    }
+  }
+
+  //---------------------------------------------------------------------
+  //---------------------------------------------------------------------
+  ::grpc::Status NiFakeService::GetAnIviDanceWithATwistStringStrlenBug(::grpc::ServerContext* context, const GetAnIviDanceWithATwistStringStrlenBugRequest* request, GetAnIviDanceWithATwistStringStrlenBugResponse* response)
+  {
+    if (context->IsCancelled()) {
+      return ::grpc::Status::CANCELLED;
+    }
+    try {
+      ViInt32 actual_size {};
+      while (true) {
+        auto status = library_->GetAnIviDanceWithATwistStringStrlenBug(0, nullptr, &actual_size);
+        if (status < 0) {
+          response->set_status(status);
+          return ::grpc::Status::OK;
+        }
+        std::string string_out;
+        if (actual_size > 0) {
+            string_out.resize(actual_size);
```

Comment:
I realize this is tricky because of how the code is structured, but it would be really great if we could manage to codegen a comment here explaining why it's not `actual_size - 1` here. If there's not a good way to do it, though, that's OK.